doc/requirements.txt

```diff
@@ -18,3 +18,5 @@ graphviz
 pybind11
 m2r2
 mistune==0.8.4
+sphinxext-opengraph
+matplotlib
```

Comment:
**Suggestion:** We should pin these to make sure the Read the Docs builds are more reproducible.
```suggestion
sphinxext-opengraph==0.9.0
matplotlib==3.8.0
```